src/Microsoft.Azure.WebJobs.Host/Config/ConcurrencyOptionsSetup.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Config
+{
+    internal class ConcurrencyOptionsSetup : IConfigureOptions<ConcurrencyOptions>
+    {
+        private const int BytesPerGB = 1024 * 1024 * 1024;
+
+        public void Configure(ConcurrencyOptions options)
+        {
+            string sku = Utility.GetWebsiteSku();
+            int numCores = Utility.GetEffectiveCoresCount();
+            Configure(options, sku, numCores);
+        }
+
+        internal static void Configure(ConcurrencyOptions options, string sku, int numCores)
+        {
+            long memoryLimitBytes = GetMemoryLimitBytes(sku, numCores);
+            if (memoryLimitBytes > 0)
+            {
+                // if we're able to determine the memory limit, apply it
+                options.TotalAvailableMemoryBytes = memoryLimitBytes;
```

Comment:
By default, the memory throttle is disabled (based on default TotalAvailableMemoryBytes default value of -1). For WebJobs/Functions running in AppService, we determine the available memory limit based on published SKU limits.